### PR TITLE
Add TransformResult as Operator::transform(...) return type

### LINKED ISSUE #650: Change Operator:transform(...) return type to be an enum
We need to be able to represent success, failure and canceled.

Related to #649

### PR DESCRIPTION
**Pull request by cjh1**
Fixes #650

### CODE REVIEW

**Comment by cquammen** [comment]
I'm getting a compiler error on Mac OS X:

```
In file included from /Users/cory.quammen/src/tomviz/tomviz/AlignWidget.cxx:19:
In file included from /Users/cory.quammen/src/tomviz/tomviz/ActiveObjects.h:24:
/Users/cory.quammen/src/tomviz/tomviz/Operator.h:46:14: error: value of type 'tomviz::OperatorState' is not implicitly convertible to 'int'
  COMPLETE = OperatorState::COMPLETE,
             ^~~~~~~~~~~~~~~~~~~~~~~
/Users/cory.quammen/src/tomviz/tomviz/Operator.h:47:14: error: value of type 'tomviz::OperatorState' is not implicitly convertible to 'int'
  CANCELED = OperatorState::CANCELED,
             ^~~~~~~~~~~~~~~~~~~~~~~
/Users/cory.quammen/src/tomviz/tomviz/Operator.h:48:11: error: value of type 'tomviz::OperatorState' is not implicitly convertible to 'int'
  ERROR = OperatorState::ERROR
          ^~~~~~~~~~~~~~~~~~~~
3 errors generated.
ninja: build stopped: subcommand failed.
```

Explicitly casting the `OperatorState` enums to int seems to solve it.

```
enum class TransformResult
{
  COMPLETE = static_cast<int>(OperatorState::COMPLETE),
  CANCELED = static_cast<int>(OperatorState::CANCELED),
  ERROR = static_cast<int>(OperatorState::ERROR)
};
```

**Comment by cquammen** [comment]
I'll push that change here if I can.

**Comment by cjh1** [comment]
@cquammen Thanks

File: tomviz/Operator.h [hunk 1]
```diff
@@ -41,6 +41,13 @@ enum class OperatorState
   ERROR
 };
 
+enum class TransformResult
+{
+  COMPLETE = static_cast<int>(OperatorState::COMPLETE),
```
Comment (cryos) [inline tomviz/Operator.h]: Would it be better to just specify the type of the enum, such as char, and cast to that, or go through the extra hoops to get at the underlying type the compiler automatically selects? This is described in item 10 of Scott Meyer's book.

**Pull request merged**